tst/com/amazon/ion/benchmark/dataGeneratorTest.java

```diff
@@ -0,0 +1,202 @@
+
+        GeneratorOptions.executeGenerator(optionsMap);
+        String format = optionsMap.get("--format").toString().substring(1, optionsMap.get("--format").toString().length() - 1);
+
+        Path path = Paths.get(optionsMap.get("<output_file>").toString());
+        byte[] buffer = Files.readAllBytes(path);
+        assertEquals(Format.valueOf(format.toUpperCase()) == Format.ION_BINARY, IonStreamUtils.isIonBinary(buffer));
+    }
+
+    /**
+     * Assert the unicode code point range of the character which constructed the generated Ion string is conform with the expect range.
+     * @throws Exception if error occurs when executing Ion data generator.
+     */
+    @Test
+    public void testGeneratedStringUniCodeRange() throws Exception {
+        Map<String, Object> optionsMap = dataGeneratorTest.testCasesGenerator("500","string", "[0,135]", "[1,4]", "test5.ion", "2021T", "[ion_text]", "[-13,-2]");
+
+        GeneratorOptions.executeGenerator(optionsMap);
+        IonReader reader = IonReaderBuilder.standard().build(new BufferedInputStream(new FileInputStream(optionsMap.get("<output_file>").toString())));
+        List<Integer> range = WriteRandomIonValues.rangeParser(optionsMap.get("--text-code-point-range").toString());
+
+        while (reader.next() != null) {
+            String str = reader.stringValue();
+            for (int i = 0; i < str.length(); i++) {
+                int codePoint = Character.codePointAt(str, i);
+                int charCount = Character.charCount(codePoint);
+                //UTF characters may use more than 1 char to be represented
+                if (charCount == 2) {
+                    i++;
+                }
+                assertTrue(codePoint >= range.get(0) && codePoint <= range.get(1));
+            }
+        }
+    }
+
+    /**
+     * Assert the generated timestamps is follow the precision and proportion of the given timestamp template.
+     * @throws Exception if error occurs when executing Ion data generator.
+     */
+    @Test
+    public void testGeneratedTimestampTemplateFormat() throws Exception{
+        Map<String, Object> optionsMap = dataGeneratorTest.testCasesGenerator("500","timestamp", "[1,135]", "[1,4]", "test6.ion", "2013T", "[ion_text]", "[-13,-2]");
+
+        ArrayList timestampsList = new ArrayList(Arrays.asList(optionsMap.get("--timestamps-template").toString().split(",")));
+
+        GeneratorOptions.executeGenerator(optionsMap);
+        IonReader reader = IonReaderBuilder.standard().build(new BufferedInputStream(new FileInputStream(optionsMap.get("<output_file>").toString())));
+        reader.next();
+
+        while (reader.isNullValue()) {
+            for ( int i = 0; i < timestampsList.size(); i++) {
+                IonReaderBuilder readerBuilder = IonReaderBuilder.standard();
+                IonReader readerTemplate = readerBuilder.build(timestampsList.get(i).toString());
```

Comment:
Consider reading the templates once into a `Timestamp` representation in-memory instead of re-reading them on each loop iteration.